test/common/http/http1/conn_pool_test.cc

```diff
@@ -293,7 +305,9 @@ TEST_F(Http1ConnPoolImplTest, VerifyAlpnFallback) {
 
   // Recreate the conn pool so that the host re-evaluates the transport socket match, arriving at
   // our test transport socket factory.
-  conn_pool_ = std::make_unique<ConnPoolImplForTest>(dispatcher_, cluster_, random_);
+  new NiceMock<Event::MockSchedulableCallback>(&dispatcher_);
```

Comment:
Just for my understanding, what triggers creation of this second SchedulableCallback?

Reply:
the constructor of conn pool calls `createSchedulableCallback`, so we need to refresh the expectation on the dispatcher
I tried to use this expectation directly, I'll try again. It's not great right now.
https://github.com/envoyproxy/envoy/blob/11dee4e9c37c244f9de4bc339fdc8695e5de2c5d/test/mocks/event/mocks.cc#L63